appstudio-controller/controllers/appstudio.redhat.com/snapshotenvironmentbinding_controller.go

```diff
@@ -192,10 +199,20 @@ func (r *SnapshotEnvironmentBindingReconciler) Reconcile(ctx context.Context, re
 			return ctrl.Result{}, nil
 		}
 
-		var err error
-		expectedDeployments[component.Name], err = generateExpectedGitOpsDeployment(component, *binding, environment, log)
-		if err != nil {
-			return ctrl.Result{RequeueAfter: time.Second * 10}, err
+		var userDevErr gitopserrors.UserError
+		expectedDeployments[component.Name], userDevErr = generateExpectedGitOpsDeployment(ctx, component, *binding, environment, rClient, log)
```

Comment:
I switched to using the `gitopserrors.UserError` package here. This allows use to report errors via .status.conditions for errors from `generateExpectedGitOpsDeployments`